apps/server/src/modules/registration/domain/service/registration.service.ts

```diff
@@ -45,4 +59,51 @@ export class RegistrationService {
 
 		return registrations;
 	}
+
+	private createRegistrationHash(): string {
+		const registrationHash = crypto.randomBytes(32).toString('base64url');
+
+		return registrationHash;
+	}
+
+	private blockForbiddenDomains(email: string): void {
+		const isBlockedDomain = isDisposableEmail(email);
+		if (isBlockedDomain) {
+			throw new BadRequestException('Registration using disposable email domains is not allowed.');
+		}
+	}
+
+	// can also be used for resending mail functionality later
+	// also think about updating registration object with resentAt timestamp
+	public generateRegistrationMail(email: string, firstName: string, lastName: string, hash: string): Mail {
```

Comment:
First, I think the generation of the registrationMail should happen in the DO itself, since all necessary information is already available there.

However if we want to include a lastSentAt timestamp and ensure the enduser doesnt get spammed, we can only ensure that here in a centralized place by moving the actual sending of the email here as well, and ensure the new timestamp is persisted after the mail was sent. (doing the timestamp stuff is in a different ticket afaik)

So 
1. move this code into DO
2. move send email code from uc here (note that this will affect module imports, @sebmue-dataport if you need help with that let me know)